org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java

```diff
@@ -1625,4 +1626,17 @@ public static boolean isReadOnly(final IResource resource) {
 		ResourceAttributes attributes = resource.getResourceAttributes();
 		return attributes != null && attributes.isReadOnly();
 	}
+
+	public static IStatus setWritable(final IFile file) {
+		ResourceAttributes attrs = file.getResourceAttributes();
+		if (attrs != null) {
+			try {
+				attrs.setReadOnly(false);
+				file.setResourceAttributes(attrs);
+			} catch (CoreException e) {
+				return e.getStatus();
```

Comment:
this does not seem to be correct. From the javadoc of `getStatus()`:

> <b>IMPORTANT:</b><br>
	 * The result must NOT be used for logging, error reporting, or as a method
	 * return value, since...